test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,252 @@
+		diff := matrixdiff.Generate(newComMatrix, docComMatrix)
+		diffStr, err := diff.String()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		err = os.WriteFile(filepath.Join(artifactsDir, "doc-diff-new"), []byte(diffFileComments+diffStr), 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("comparing new and documented commatrices")
+		areEqual := comMatricesAreEqual(*newComMatrix, *docComMatrix)
+		o.Expect(areEqual).To(o.BeTrue())
+	})
+})
+
+// getClusterXYersion return cluster's version in X.Y stream
+func getClusterXYersion(configClient *configv1client.ConfigV1Client) (string, error) {
+	clusterVersion, err := configClient.ClusterVersions().Get(context.Background(), "version", metav1.GetOptions{})
+	if err != nil {
+		return "", nil
+	}
+	ClusterVersionParts := strings.SplitN(clusterVersion.Status.Desired.Version, ".", 3)
+	if len(ClusterVersionParts) < 2 {
+		return "", fmt.Errorf("Cluster Version has only X stream.")
+	}
+	return strings.Join(ClusterVersionParts[:2], "."), nil
+}
+
+// isSNOCluster will check if OCP is a single node cluster
+func isSNOCluster(oc *configv1client.ConfigV1Client) (bool, error) {
+	infrastructureType, err := oc.Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	logrus.Infof("the cluster type is %s", infrastructureType.Status.ControlPlaneTopology)
+	return infrastructureType.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode, nil
+}
+
+func isBMCluster(oc *configv1client.ConfigV1Client) (bool, error) {
+	infrastructureType, err := oc.Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	logrus.Infof("the cluster platform is %s", infrastructureType.Status.PlatformStatus.Type)
+	return infrastructureType.Status.PlatformStatus.Type == configv1.BareMetalPlatformType, nil
+}
+
+// createCSVFromUrl creates a CSV from the given URL at fp filepath
+func createCSVFromUrl(url string, fp string) error {
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	file, err := os.Create(fp)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// comMatricesAreEqual return true if given comMatrices are equal, and false otherwise
+func comMatricesAreEqual(cm1 types.ComMatrix, cm2 types.ComMatrix) bool {
+	diff1 := cm1.Diff(cm2)
+	diff2 := cm2.Diff(cm1)
+
+	// Check if the Diff matrices are not empty
+	if len(diff1.Matrix) > 0 || len(diff2.Matrix) > 0 {
+		return false
+	}
+
+	// Diff matrices are empty, ComMatrices are equal
+	return true
+}
+
+// excludeBareMetalEntries excludes and returns only comDetails that are not bare metal static entries
+func excludeGivenStaticEntries(comDetails []types.ComDetails, masterStaticEntriesMatrix, workerStaticEntriesMatrix *types.ComMatrix) []types.ComDetails {
```

Comment:
I think this is confusing, it gets master and worker entries but in master for example you send StandardStaticEntries which have both master and worker. I think this can be a func that gets a matrix and a list of entries to ignore/remove (checks the role match), can also be added in the matrix repo